xapian-letor/api/featurevector.cc

```diff
@@ -48,14 +48,33 @@ using namespace std;
 
 using namespace Xapian;
 
-FeatureVector::FeatureVector() {
+class FeatureVector::Feature_internal : public Xapian::Internal::intrusive_base
+{
+  	friend class FeatureVector;
+	double label;
+	std::map<int,double> fvals;
+	int fcount;
+	string did;
+
+  public:
+  	map<string, map<string, int> > load_relevance(const std::string & qrel_file);
+};
+
+FeatureVector::FeatureVector() : internal(new FeatureVector::Feature_internal) { }
+
+FeatureVector::FeatureVector(const FeatureVector & o) : internal(o.internal) { }
```

Comment:
We'd usually format these with the braces on their own lines (see `api/omdocument.cc`, for instance).